### DISCUSSION TITLE
Livewire 2: Attempting to validate array of objects wipes out the values

### DISCUSSION THREAD

**Discussion by MartinHughes-BPC** [livewire/livewire#6890] [1. Help]
Hi all,

I'm not sure if this is a bug or intended. I've created a Livewire playground to demonstrate the problem here: https://laravelplayground.com/#/snippets/5db4980d-2256-47d1-9063-bbf268486d1b

I get a set of grades from the database and set them to the $grades public property.

In the view I loop through the grades and  spit out an input for each grade and provide a save button, which calls a 'save' method on the component. I use binding to hook up the inputs to the backend.

In the component, because I want these grades as objects again and not the arrays that Livewire hydrates them as, I use the hydrateGrades method and convert them back to objects. (Makes it easier between Livewire calls, because you can always use Object->Property syntax). 

I have the following $rules: $rules = ['grades' => 'nullable|max:10'].

When I click the 'save' button the validation appears to run, however it wipes out all the grade values and sets them to null!

This only seems to affect arrays of objects - if I instead convert the DB objects to arrays, get rid of my hydrate method, and use array syntax everywhere it all works as expected.

Any thoughts?

Cheers,
Martin.

**Comment by andysh-uk** [comment]
Hi @MartinHughes-BPC 

I've come across the exact same problem in an app I'm writing with L11 and LW3.

In my case, I'm using the `Wireable` interface rather than the hydrate methods, but I see the same.

After validation, all properties of my custom objects are wiped out (if I do a `dd` just before the validation call, they are all there.)

If I remove the validation rules for my array of objects, I don't see the problem.

I have a similar custom object which is a singular object (not an array of them) and this seems to work fine.

Did you ever get a solution?  My only thought is to convert everything to array syntax and get rid of my custom object, which just isn't as nice!
